service_container/tags.rst

```diff
@@ -564,3 +564,241 @@ application handlers.
+                arguments: [!tagged { tag: 'app.handler', index_by: 'index_attribute_name' }]
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="App\Handler\One">
+                    <tag name="app.handler" index_attribute_name="hOne" />
+                </service>
+
+                <service id="App\Handler\Two">
+                    <tag name="app.handler" index_attribute_name="hTwo" />
+                </service>
+
+                <service id="App\HandlerCollection">
+                    <!-- inject all services tagged with app.handler as first argument -->
+                    <argument type="tagged" tag="app.handler" index-by="index_attribute_name"/>
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+
+        $container->register(App\Handler\One::class)
+            ->addTag('app.handler', ['index_attribute_name' => 'hOne']);
+
+        $container->register(App\Handler\Two::class)
+            ->addTag('app.handler', ['index_attribute_name' => 'hTwo']);
+
+        $container->register(App\HandlerCollection::class)
+            // inject all services tagged with app.handler as first argument
+            ->addArgument(new TaggedIteratorArgument('app.handler', 'index_attribute_name'));
+
+After compilation the ``HandlerCollection`` service is able to iterate over your
+application handlers. But we want to retrieve a specific service by it's
+``index_attribute_name``. As it's an iterator, we can use ``iterator_to_array``
+to get an array and then retrieve the ``hTwo`` handler :
+
+.. code-block:: php
+
+    // src/HandlerCollection.php
+    namespace App;
+
+    class HandlerCollection
+    {
+        public function __construct(iterable $handlersIt)
+        {
+            $handlers = iterator_to_array($handlersIt);
+
+            $hTwo = $handlers['hTwo']:
+            $hOne = $handlers['hOne'];
+        }
+    }
+
+
+.. tip::
+
+    You can omit the ``index_attribute_name`` attribute, to do this you
+    have, by default, to implement a static method named like this : ``getDefaultIndexAttributeNameName``
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # config/services.yaml
+            services:
+                App\Handler\One:
+                    tags:
+                        - { name: 'app.handler', priority: 20 }
+
+            App\Handler\Two:
+                tags:
+                    - { tag: 'app.handler', index : 'hTwo' }
```

Comment:
should be „name:“ I guess and please remove the space after Index before the colon